### PR TITLE
Remove mocks

### PR DESCRIPTION
**Pull request by edeyoung**
Remove mocks so that dealing with database changes is easier. Mocks unnecessary - can fulfill testing needs with integration tests and forcing errors naturally.

Key changes:
* made all unit tests that use mocks into integration tests that use a db connection and force db errors by making entries too long.
* moved the capability querier test that goes out and queries the capability statements into the e2e tests.
  * made it so this only runs if the chplapikey is set. can't set this on travis because it's a private key, so this test is not run on travis.
* made e2e tests setup and breakdown the db (including removing all entries after the run is complete) so that we can do repeated runs with the integration tests as well.
  * this made it so there's a global `store` variable, so removed in-function creations of the `store` and only use this global one
* made it so the travis config runs two travis "stages". this is because the integration tests run in the terminal and need to use `localhost` as the hostname for the queue. however, the e2e test needs to use the queue container name for the host because it runs in a container. Also:
  * updated travis to go 1.14
  * made environment variables global where appropriate
  * runs the integration tests after `wait-for-it` passes ensuring all container connections are available
* updated how travis handles linting
  * with update to go 1.14, can no longer recursively lint over everything. now need to lint in every go package.
  * created a make command to do this.
  * called the make command in travis config
  * commented out networkstatsquerier lint because a weird error is occurring only in travis for this.
----

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-181
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	}
+
+	code := m.Run()
+
+	teardown()
+	os.Exit(code)
+}
+
+func Test_saveMsgInDB(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
```
Comment (bmath10) [inline endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go:80]: It looks like here as well as in productquerier_integration_test, populatefhirendpoints_integration_test, and capabilityinfo_integration_test you could just use the new th.IntegrationDBTestSetupMain in the TestMain for those files, unless there's something I'm missing

File: endpointmanager/pkg/chplmapper/capabilityinfo_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+
+	dstu2, err = capabilityparser.NewCapabilityStatement(dstu2JSON)
+	th.Assert(t, err == nil, err)
+
+	vendor, err = getVendorMatch(ctx, dstu2, store)
+	th.Assert(t, err != nil, "expected error due to accessing the copyright")
+	th.Assert(t, len(vendor) == 0, fmt.Sprintf("expected no vendor value. Instead got %s", vendor))
+
+	// allscripts
+	expected = "Allscripts" // the capability statement is missing the publisher
```
Comment (bmath10) [inline endpointmanager/pkg/chplmapper/capabilityinfo_integration_test.go:295]: this comment and the one below for meditech seem incorrect?

File: endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints_integration_test.go [hunk 3]
```diff
@@ -83,6 +87,141 @@ func Test_Integration_AddEndpointData(t *testing.T) {
...
+func Test_AddEndpointData(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
+	defer teardown(t, store.DB)
+
+	var err error
+
+	endpt1 := testEndpointEntry
+	endpt2 := testEndpointEntry
+	// fhirEndpt := testFHIREndpoint
+	// var savedEndpt *endpointmanager.FHIREndpoint
```
Comment (bmath10) [inline endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints_integration_test.go:167]: these comments look like they can be deleted

File: capabilityquerier/pkg/capabilityquerier/capabilityquerier_integration_test.go [hunk 4]
```diff
@@ -93,7 +93,7 @@ func Test_Integration_GetAndSendCapabilityStatement(t *testing.T) {
 	th.Assert(t, err == nil, err)
 	// need to pause to ensure all messages are on the queue before we count them
 	time.Sleep(10 * time.Second)
-	th.Assert(t, count == 10, fmt.Sprintf("expected there to be 10 messages in the queue; saw %d", count))
+	th.Assert(t, count >= 9, fmt.Sprintf("expected there to be 9 or 10 messages in the queue (difference because dealing with real data and endpoints); saw %d", count))
```
Comment (mayerm94) [inline capabilityquerier/pkg/capabilityquerier/capabilityquerier_integration_test.go:96]: (count == 9 || count == 10)

File: endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go (hunk 1 above)
Comment (edeyoung) [reply endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go:80]: the setup ensure the db is empty. the teardown clears out the db.
in the integration tests, we want a clean db for each individual test.
however, the way the e2e test is written, it depends on data that was previously entered in the db. so the integration tests do it on a test-by-test basis, and the e2e does it once in main.

Comment (bmath10) [reply endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go:80]: oh okay makes sense!

**Comment by bmath10** [comment]
I think one of the CHPL URLs is occasionally breaking, since I'm inconsistently seeing this error show up while running tests:
```
Test_GetCHPLProducts: data_flow_test.go:63: Test_GetCHPLProducts: context deadline
exceeded (Client.Timeout or context cancellation while reading body)
lantern-e2e                      |         reading the CHPL response body failed
```
I don't know if there's anything we want to do about it right now but I figured I'd mention it!

**Pull request merged**